optuna/integration/allennlp.py

```diff
@@ -103,14 +103,22 @@ def _build_params(self) -> Dict[str, Any]:
         """Create a dict of params for AllenNLP."""
         # _build_params is based on allentune's train_func.
         # https://github.com/allenai/allentune/blob/master/allentune/modules/allennlp_runner.py#L34-L65
-        for key, value in self._params.items():
-            self._params[key] = str(value)
-        _params = json.loads(_jsonnet.evaluate_file(self._config_file, ext_vars=self._params))
+        params = self._environment_variables()
```

Comment:
Could you add a test case about the environmental variables? This change affects the behavior of a public method, and I think it would prevent degradation during future development.

Reply:
Added in c5e96a9.